chain/indexer/src/streamer/streamer.rs

```diff
@@ -194,11 +197,34 @@ async fn fetch_chunks(
 pub(crate) async fn start(
     view_client: Addr<near_client::ViewClientActor>,
     client: Addr<near_client::ClientActor>,
-    mut blocks_sink: mpsc::Sender<BlockResponse>,
+    indexer_config: IndexerConfig,
+    mut blocks_sink: mpsc::Sender<StreamerMessage>,
 ) {
     info!(target: INDEXER, "Starting Streamer...");
+    let mut indexer_db_path = neard::get_store_path(&indexer_config.home_dir);
+    indexer_db_path.push_str("/indexer");
+
+    // TODO: implement proper error handling
+    let db = DB::open_default(indexer_db_path).unwrap();
     let mut outcomes_to_get = Vec::<types::TransactionOrReceiptId>::new();
-    let mut last_synced_block_height: types::BlockHeight = 0;
+    let mut last_synced_block_height: Option<types::BlockHeight> = match indexer_config.sync_mode {
+        crate::SyncModeEnum::FromInterruption => match db.get(b"last_synced_block_height") {
+            Ok(Some(value)) => Some(String::from_utf8(value).unwrap().parse::<u64>().unwrap()),
+            Ok(None) => None,
+            Err(_) => {
+                db.put(b"last_synced_block_height", b"0").unwrap();
+                None
+            }
+        },
+        crate::SyncModeEnum::LatestSynced => None, // let streamer to handle from real-time
+        crate::SyncModeEnum::BlockHeight(height) => Some(height.checked_sub(1).unwrap_or(0)), // provided block inclusively, // provided block inclusively
```

Comment:
Well, on the second thought, `BlockHeight(0)` and `BlockHeight(1)` are now treated the same and it will skip the block height 0.

P.S. the comment message is duplicated.